### PR TITLE
Add more tests to cover the subset of datasets

### PR DESCRIPTION
**Pull request by zhangguanheng66**
This PR includes:
- Add more tests to cover the scenario for the subset of datasets

### CODE REVIEW

File: test/data/test_builtin_datasets.py [hunk 1]
```diff
@@ -89,9 +83,13 @@ def test_penntreebank(self):
         tokens_ids = [vocab[token] for token in 'the player characters rest'.split()]
         self.assertEqual(tokens_ids, [2, 2550, 3344, 1125])
 
+        # Add test for the subset of the standard datasets
+        train_dataset, test_dataset = PennTreebank(data_select=('train', 'test'))
+        train_iter, test_iter = torchtext.experimental.datasets.raw.PennTreebank(data_select=('train', 'test'))
```
Comment (cpuhrsch) [inline test/data/test_builtin_datasets.py:88]: I'd check whether the returned iterator then is also indeed for training or testing as you'd expect. That way we make sure this subset selection is working.

Comment (zhangguanheng66) [reply test/data/test_builtin_datasets.py:106]: Fixed

File: test/data/test_builtin_datasets.py [hunk 2]
```diff
@@ -84,14 +89,29 @@ def test_penntreebank(self):
...
 
         vocab = train_dataset.get_vocab()
         tokens_ids = [vocab[token] for token in 'the player characters rest'.split()]
         self.assertEqual(tokens_ids, [2, 2550, 3344, 1125])
 
+        # Add test for the subset of the standard datasets
+        train_dataset, test_dataset = PennTreebank(data_select=('train', 'test'))
+        self.assertEqual(len(train_dataset), 924412)
+        self.assertEqual(len(test_dataset), 82114)
+        self.assertEqual(train_dataset[20:25], torch.tensor([9919, 9920, 9921, 9922, 9188]).long())
```
Comment (cpuhrsch) [inline test/data/test_builtin_datasets.py:104]: nit: [torch.tensor](https://pytorch.org/docs/master/generated/torch.tensor.html#torch.tensor) has a dtype flag, so you can write ```torch.tensor([...], dtype=torch.int64)```.

File: test/data/test_builtin_datasets.py [hunk 3]
```diff
@@ -56,11 +46,26 @@ def test_wikitext2(self):
...
 
         vocab = train_dataset.get_vocab()
         tokens_ids = [vocab[token] for token in 'the player characters rest'.split()]
         self.assertEqual(tokens_ids, [2, 286, 503, 700])
 
+        # Add test for the subset of the standard datasets
+        train_dataset, test_dataset = torchtext.experimental.datasets.raw.WikiText2(data_select=('train', 'test'))
+        next(iter(train_dataset))
+        self.assertEqual(next(iter(train_dataset)), ' = Valkyria Chronicles III = \n')
+        next(iter(test_dataset))
```
Comment (cpuhrsch) [inline test/data/test_builtin_datasets.py:61]: based on offline discussion: Another way to get rid of the iterator that's being purposefully overwritten by this naming is to use ```del train_iter```

File: test/data/test_builtin_datasets.py [hunk 4]
```diff
@@ -56,11 +46,26 @@ def test_wikitext2(self):
...
+        self.assertEqual(train_dataset[20:25], torch.tensor([5024, 89, 21, 3, 1838]).long())
+        self.assertEqual(test_dataset[30:35], torch.tensor([914, 4, 36, 11, 569]).long())
+        self.assertEqual(valid_dataset[40:45], torch.tensor([925, 8, 2, 150, 8575]).long())
 
         vocab = train_dataset.get_vocab()
         tokens_ids = [vocab[token] for token in 'the player characters rest'.split()]
         self.assertEqual(tokens_ids, [2, 286, 503, 700])
 
+        # Add test for the subset of the standard datasets
+        train_dataset, test_dataset = torchtext.experimental.datasets.raw.WikiText2(data_select=('train', 'test'))
```
Comment (cpuhrsch) [inline test/data/test_builtin_datasets.py:58]: You can also add a length test to this.

File: test/data/test_builtin_datasets.py [hunk 5]
```diff
@@ -124,13 +154,31 @@ def test_imdb(self):
...
+        self.assertEqual(text[:25], 'I rented I AM CURIOUS-YEL')
+
     def test_multi30k(self):
         from torchtext.experimental.datasets import Multi30k
         # smoke test to ensure multi30k works properly
         train_dataset, valid_dataset, test_dataset = Multi30k()
         self.assertEqual(len(train_dataset), 29000)
         self.assertEqual(len(valid_dataset), 1014)
         self.assertEqual(len(test_dataset), 1000)
+        self.assertEqual(train_dataset[20], (torch.tensor([3, 443, 2530, 46, 17478, 7422, 7, 157, 9, 11, 5848, 2]).long(),
```
Comment (cpuhrsch) [inline test/data/test_builtin_datasets.py:176]: ```:%s/torch.tensor([\(.*\)]).long()/torch.tensor(\[\1\], dtype=torch.int64/g```
vim regex search replace pattern to save you some time.

File: test/data/test_builtin_datasets.py [hunk 6]
```diff
@@ -18,13 +18,12 @@ def conditional_remove(f):
 
 
 class TestDataset(TorchtextTestCase):
+    def helper_test_func(self, length, target_length, results, target_results):
```
Comment (cpuhrsch) [inline test/data/test_builtin_datasets.py:21]: Using the [style guide](https://www.python.org/dev/peps/pep-0008/#designing-for-inheritance) we should probably prefix this with a leading underscore.

**Review by cpuhrsch — APPROVED** [review]

File: test/data/test_builtin_datasets.py (hunk 6 above)
Comment (cpuhrsch) [reply test/data/test_builtin_datasets.py:21]: As a suggestion, you could have one helper for datasets and one for iterators if you want to distinguish between these two cases to make writing code easier. You could also move the tensor construction ```torch.tensor([...], dtype=torch.int64)``` into these helpers.

**Pull request merged**